// react-contexify/lib/Component/ProxyContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _eventManager = require('./../util/eventManager');

var _eventManager2 = _interopRequireDefault(_eventManager);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ProxyContainer = function (_Component) {
  _inherits(ProxyContainer, _Component);

  function ProxyContainer(props) {
    _classCallCheck(this, ProxyContainer);

    var _this = _possibleConstructorReturn(this, (ProxyContainer.__proto__ || Object.getPrototypeOf(ProxyContainer)).call(this, props));

    _this.proxyRender = function (element) {
      return _this.setState(function (prevState) {
        return { element: [].concat(_toConsumableArray(prevState.element), [element]) };
      });
    };

    _this.proxyUnmount = function (menuKey) {
      return _this.setState(function (prevState) {
        return { element: prevState.filter(function (menu) {
            return menu.key !== menuKey;
          }) };
      });
    };

    _this.state = {
      element: []
    };
    return _this;
  }

  _createClass(ProxyContainer, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      _eventManager2.default.on('PROXY_RENDER', this.proxyRender);
      _eventManager2.default.on('PROXY_UNMOUNT', this.proxyUnmount);
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      _eventManager2.default.off('PROXY_RENDER');
      _eventManager2.default.off('PROXY_UNMOUNT');
    }
  }, {
    key: 'renderChildren',
    value: function renderChildren() {
      return this.state.element.map(function (el) {
        return el;
      });
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        null,
        this.renderChildren()
      );
    }
  }]);

  return ProxyContainer;
}(_react.Component);

exports.default = ProxyContainer;
