// react-contexify/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ContextMenu = exports.IconFont = exports.menuProvider = exports.ContextMenuProvider = exports.Separator = exports.Item = undefined;

var _Item = require('./Component/Item');

Object.defineProperty(exports, 'Item', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Item).default;
  }
});

var _Separator = require('./Component/Separator');

Object.defineProperty(exports, 'Separator', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Separator).default;
  }
});

var _ContextMenuProvider = require('./Component/ContextMenuProvider');

Object.defineProperty(exports, 'ContextMenuProvider', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_ContextMenuProvider).default;
  }
});

var _menuProvider = require('./menuProvider');

Object.defineProperty(exports, 'menuProvider', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_menuProvider).default;
  }
});

var _IconFont = require('./Component/IconFont');

Object.defineProperty(exports, 'IconFont', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_IconFont).default;
  }
});

var _ContextMenu = require('./Component/ContextMenu');

var _ContextMenu2 = _interopRequireDefault(_ContextMenu);

var _withProxy = require('./util/withProxy');

var _withProxy2 = _interopRequireDefault(_withProxy);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Ctx = (0, _withProxy2.default)(_ContextMenu2.default);

exports.ContextMenu = Ctx;
